src/test/func/malloc/malloc.cc

```diff
@@ -229,6 +265,25 @@ int main(int argc, char** argv)
     test_posix_memalign(0, align + 1, EINVAL, true);
   }
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+  void* a = nullptr;
+  test_reallocarr(a, 1, 0, SUCCESS, false);
+
+  for (smallsizeclass_t sc = 0; sc < (MAX_SMALL_SIZECLASS_BITS + 4); sc++)
+  {
+    const size_t size = bits::one_at_bit(sc);
+    void* p = nullptr;
+    test_reallocarr(p, 1, size, SUCCESS, false);
```

Comment:
I still don't get the purpose of `p` here?  It is set to `nullptr` and only passed by value, so it is never modified.